### PR TITLE
Feat/integrate new bylines

### PR DESCRIPTION
**Pull request by nehasri89**
This PR integrates the new `bylines` field as introduced as part of 
[TPA PR ](https://github.com/newsuk/times-public-api/pull/850) to TC

This cannot be merged until the TPA PR is merged in and the migration of data has happened.

### CODE REVIEW

File: packages/article-magazine-comment/src/article-magazine-comment.js [hunk 1]
```diff
@@ -40,11 +37,20 @@ class ArticleMagazineComment extends Component {
...
+        ? bylines[0].image.crop.url
+        : null;
+
     return (
       <Fragment>
         <ArticleHeader
-          authorImage={author.image}
-          byline={byline}
+          authorImage={authorImage}
+          byline={bylines[0].byline}
```
Comment (tuncaulubilge) [inline packages/article-magazine-comment/src/article-magazine-comment.js]: We could use another `bylines.length>0` check here

File: packages/article-main-comment/src/article-main-comment.js [hunk 2]
```diff
@@ -31,12 +30,19 @@ class ArticlePage extends Component {
...
+      Object.keys(bylines[0].image).length !== 0 &&
+      bylines[0].image.crop
+        ? bylines[0].image.crop.url
+        : null;
 
     return (
       <ArticleHeader
         authorImage={authorImage}
-        byline={byline}
+        byline={bylines[0].byline}
```
Comment (tuncaulubilge) [inline packages/article-main-comment/src/article-main-comment.js]: Same as the other templates, this need to check if bylines is not null or empty first

File: packages/article-main-comment/src/article-main-comment.web.js [hunk 3]
```diff
@@ -30,10 +27,19 @@ class ArticlePage extends Component {
...
+      bylines[0].image.crop
+        ? bylines[0].image.crop.url
+        : null;
+
     return (
       <ArticleHeader
-        authorImage={author.image}
-        byline={byline}
+        authorImage={authorImage}
+        byline={bylines[0].byline}
```
Comment (tuncaulubilge) [inline packages/article-main-comment/src/article-main-comment.web.js]: .

File: packages/article-magazine-comment/src/article-magazine-comment.web.js [hunk 4]
```diff
@@ -37,11 +34,20 @@ class ArticlePage extends Component {
...
+        ? bylines[0].image.crop.url
+        : null;
+
     return (
       <Fragment>
         <ArticleHeader
-          authorImage={author.image}
-          byline={byline}
+          authorImage={authorImage}
+          byline={bylines[0].byline}
```
Comment (tuncaulubilge) [inline packages/article-magazine-comment/src/article-magazine-comment.web.js]: .

File: packages/provider-queries/src/article.js [hunk 5]
```diff
@@ -4,6 +4,21 @@ import gql from "graphql-tag";
 export default addTypenameToDocument(gql`
   query ArticleQuery($id: ID!) {
     article(id: $id) {
+      bylines {
+        ... on TextByline {
```
Comment (jmars) [inline packages/provider-queries/src/article.js]: Should be ... on Byline, the interface shared between AuthorByline and TextByline

File: packages/provider-queries/src/article.web.js [hunk 6]
```diff
@@ -4,6 +4,21 @@ import gql from "graphql-tag";
 export default addTypenameToDocument(gql`
   query ArticleQuery($id: ID!) {
     article(id: $id) {
+      bylines {
+        ... on TextByline {
```
Comment (jmars) [inline packages/provider-queries/src/article.web.js]: Same as above

File: packages/article-main-comment/src/article-main-comment.js [hunk 7]
```diff
@@ -31,12 +30,24 @@ class ArticlePage extends Component {
...
+    const authorByline =
+      bylines && bylines.length > 0 && Object.keys(bylines[0]).length !== 0
+        ? bylines[0].byline
+        : null;
 
     return (
       <ArticleHeader
         authorImage={authorImage}
-        byline={byline}
+        byline={authorByline}
```
Comment (tuncaulubilge) [inline packages/article-main-comment/src/article-main-comment.js]: At the moment, byline doesn't support null values, which causes a red screen on these templates. The article page shouldn't crash when we don't have any bylines.

Comment (aronshamash) [reply packages/article-main-comment/src/article-main-comment.js]: added null checks in `article-meta` components now.

Comment (tuncaulubilge) [reply packages/article-main-comment/src/article-main-comment.js]: I'll give it a try and approve afterwards

**Pull request merged**